clientd/src/lib.rs

```diff
@@ -12,18 +14,55 @@ use tokio::sync::{broadcast, mpsc, oneshot};
 mod rpc;
 pub mod server;
 
-// TODO: remove pub and use this type to enforce invariants
-#[derive(Debug, Serialize)]
-pub struct CallbackResponse(pub Result<String, RpcError>);
+#[derive(Clone)]
+pub struct EventSubscribers(HashMap<EventKey, broadcast::Sender<Event>>);
 
-// TODO: figure out best approach for error handling
-#[derive(Debug, Serialize)]
-pub enum RpcError {
-    ClientError,
+impl EventSubscribers {
+    pub fn new() -> Self {
+        let mut se_map = HashMap::new();
+        for event in EventKey::iter() {
+            let (tx, _rx) = broadcast::channel(128);
+            se_map.insert(event, tx);
+        }
+        Self(se_map)
+    }
+}
+
+impl Default for EventSubscribers {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl AsRef<HashMap<EventKey, broadcast::Sender<Event>>> for EventSubscribers {
+    fn as_ref(&self) -> &HashMap<EventKey, broadcast::Sender<Event>> {
+        &self.0
+    }
+}
+
+pub struct FedimintClient(Client<UserClientConfig>);
+
+impl FedimintClient {
+    pub async fn new(cfg: PathBuf) -> Self {
+        let cfg_path = cfg.join("client.json");
+        let db_path = cfg.join("client.db");
+        let cfg: UserClientConfig = load_from_file(&cfg_path);
+        let db = fedimint_rocksdb::RocksDb::open(db_path).unwrap().into();
+
+        let client = Client::new(cfg.clone(), db, Default::default()).await;
+        Self(client)
+    }
+}
+
+impl AsRef<Client<UserClientConfig>> for FedimintClient {
+    fn as_ref(&self) -> &Client<UserClientConfig> {
+        &self.0
+    }
 }
+
 #[derive(Debug)]
-pub enum ManagerMessage {
-    CallHandler(RpcRequest, oneshot::Sender<CallbackResponse>),
+pub enum DispatcherMessage {
```

Comment:
Both previous and new name seems sus. Docstring would be great because on the first look I don't even know what it is. :)

Reply:
`DispatcherMessage` does  not clarify ... is it *from* dispatcher, *to* dispatcher, *by* dispatcher?

Reply:
(it is _for_ dispatcher) yes I need to do more doc strings. should I do it this PR or can I just do it in the next one ?